python/cucim/src/cucim/core/operations/spatial/rotate_and_flip.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+
+import cupy
+import numpy as np
+import scipy.ndimage as ndimage
+
+
+def image_flip(
+    img: Any, 
+    spatial_axis: tuple()
+    )-> Any:
+    """
+    Shape preserving order reversal of elements in input array along the given spatial axis
+
+    Parameters
+    ----------
+    img : cupy.ndarray or numpy.ndarray
+        Input data. Can be numpy.ndarray or cupy.ndarray
+    spatial_axis : tuple
+        spatial axes along which to flip over the input array
+        
+    Returns
+    -------
+    out : cupy.ndarray or numpy.ndarray
+        Output data. Same dimensions and type as input.
+
+    Raises
+    ------
+    TypeError
+        If input 'img' is not cupy.ndarray or numpy.ndarray
+    
+    Examples
+    --------
+    >>> import cucim.core.operations.spatial as spt
+    >>> # input is channel first 3d array
+    >>> output_array = spt.image_flip(input_arr,(1,2))
+    """
+    try:
+        iscupy = False
+        cupy_img = img
+        if isinstance(img, np.ndarray):
+            iscupy = True
+            cupy_img = cupy.asarray(img.astype(img.dtype))    
```

Comment:
```suggestion
            cupy_img = cupy.asarray(img)
```
`cupy.flip` will not require ensuring contiguity of the input